### PR TITLE
[LOCKSTEP UPGRADE #45525] stop using Instance::def_ty

### PR DESCRIPTION
**Pull request by arielb1**
that function has been removed

### CODE REVIEW

File: miri/lib.rs [hunk 1]
```diff
@@ -157,7 +158,7 @@ pub fn eval_main<'a, 'tcx: 'a>(
         Ok(())
     }
 
-    let mut ecx = EvalContext::new(tcx, ty::ParamEnv::empty(traits::Reveal::All), limits, Default::default(), Default::default());
+    let mut ecx = EvalContext::new(tcx, ty::ParamEnv::empty(traits::Reveal::All), limits, Default::default(), Default::default(), Substs::empty());
```
Comment (arielb1) [inline miri/lib.rs]: This does not compile in practice, but it's not my commit.

Comment (arielb1) [reply miri/lib.rs]: cc @oli-obk - that's your commit

Comment (eddyb) [reply miri/lib.rs]: Is this a `rustc` master vs nightly difference?

Comment (oli-obk) [reply miri/lib.rs]: It's https://github.com/solson/miri/tree/upstreamed2 which hasn't been merged in this repo yet. The corresponding PR was merged yesterday

**Pull request merged**